modules/ingester/instance.go

```diff
@@ -80,7 +86,9 @@ func newInstance(instanceID string, limiter *Limiter, writer tempodb.Writer) (*i
 		limiter:            limiter,
 		writer:             writer,
 
-		hash: fnv.New32(),
+		waitForFlush: atomic.NewInt32(0),
+		flushChan: make(chan int, 10), // should not need more than 2
```

Comment:
i don't think we need this and am scared of tracking state like this.  we should be able to just check the length of the flush op queue to determine if additional work needs to be done.

Reply:
unfortunately in the time between de-queuing a "complete" operation and queuing a "flush" operation, the queue will appear empty as @mdisibio pointed out.. maybe there's some things we can do there